apps/screengrab/screengrab-1.py

```diff
@@ -245,7 +245,7 @@ def _run( self, args ) :
 		if args["command"].value :
 			exec( args["command"].value, d, d )
 		if args["commandFile"].value :
-			execfile( args["commandFile"].value, d, d )
+			exec( open( args["commandFile"].value ).read(), d, d )
```

Comment:
I believe this loses useful information like the filename (and possibly line number) from any exceptions thrown. I believe the "blessed" way of doing this in Python 3 is this : https://github.com/ImageEngine/cortex/blob/master/python/IECore/ConfigLoader.py#L76-L80. Why they removed `execfile()` I don't know.

Reply:
That's concise isn't it! Amended [here](https://github.com/GafferHQ/gaffer/pull/3898/commits/3408f7140eacc6c482eb909373f5e114f42a5268#diff-ada6111e65cc5bf50c955e4d33b98c73L248-R252). Thanks.

Reply:
> That's concise isn't it!

I watched Herb Sutter's [BridgeToNewThingia](https://herbsutter.com/2020/06/14/talk-video-available-bridge-to-newthingia-devaroundthesun/) talk yesterday. Summary :

- Getting people to adopt new things is hard.
- If you are the incumbent, then your competition is your previous release.
- If you are not compatible with your competition, you may still succeed, but you will set back adoption approximately one decade.

Python 3 is mentioned.